core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -346,11 +346,7 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
   }
 
   def getWorkspaceContextAndPermissions(workspaceName: WorkspaceName, requiredAction: SamResourceAction): Future[SlickWorkspaceContext] = {
-    dataSource.inTransaction { dataAccess =>
-      withWorkspaceContextAndPermissions(workspaceName, requiredAction, dataAccess) { workspaceContext =>
-        DBIO.successful(workspaceContext)
-      }
-    }
+    withWorkspaceContextAndPermissionsF(workspaceName, requiredAction) { Future(_) }
   }
```

Comment:
we may want to rename `getWorkspaceContextAndPermissions` to something more accurate (it checks permissions, but doesn't return them). At least in this draft I'm not renaming.